packages/app-builder/src/routes/ressources+/data+/editField.tsx

```diff
@@ -78,11 +85,71 @@ export async function action({ request }: ActionFunctionArgs) {
   }
 }
 
+type fieldIsUniqueOverride = {
+  checked: boolean;
+  disabled: boolean;
+  reason:
+    | 'cannot_toggle_enum_enabled'
+    | 'cannot_toggle_index_pending'
+    | 'cannot_untoggle_field_linked'
+    | 'object_id_must_be_unique'
+    | '';
+};
```

Comment:
To remove.
In that particular type of usecase, recommandation is to let Typescript infer return type (espectially when this is the return type of a local function)

Reply:
alright 👍 
This felt right because it made for proper autocomplete where I was using it below but fine

Reply:
but please tell me again, when is the convention to have an explicit output type vs inferred ? When you don't need to verify that your helper function outputs a specific type ?

Reply:
In general, TS recommandations advise to let TS infer return type of function. In a lot of cases, it does the job better than an explicit declaration (by doing more discrimination + literral types where people tends to define a `string` for example).

Still, there are some exceptions: defining a public interface which in the end will be used outside of the module, in another pkg / file / etc... can be a good idea, since this is the public contract we ensure with the consumer (so in this special case, we want TS to ensure we respect the contract in our implementation).

Basically the question is more or less this one in my mental model :
- I want TS to check explicitly I respect the public contract I defined for this module: explicit types
- I want TS to check the integration of my function: implicit types (TS does a better job at it + VS Code allow autocomplete so no DX lost)

Reply:
alright thanks!